### PR TITLE
unify normalize path function and improve its performance

### PR DESCRIPTION
**Pull request by janfjohannes**
unify normalize path function and improve its performance

---
[//]: # (BEGIN SAPLING FOOTER)
* #2924
* #2820
* __->__ #2894

### CODE REVIEW

**Comment by jldec** [comment]
related
https://github.com/opral/monorepo/issues/2184
LIX-62

**Review by martin-lysk — APPROVED** [review]
LGTM - just one comment for clearification

File: lix/packages/fs/src/utilities/helpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+})
+
+describe("keep trailing slashes", () => {
+	const units: [string, string][] = [
+		["\\", "/"],
+ 		["foo\\bar\\baz\\", "/foo/bar/baz/"],
+	 	["foo\\\\bar\\\\baz\\\\", "/foo/bar/baz/"],
+		["foo//bar//baz//", "/foo/bar/baz/"],
+		["foo/bar/baz/", "/foo/bar/baz/"],
+		["./foo/bar/baz/", "/foo/bar/baz/"],
```
Comment (felixhaeberle) [inline lix/packages/fs/src/utilities/helpers.test.ts:57]: Can we aslso get an example with `./` in the middle of a path? I don't see this tested here, only `../`.

**Review by felixhaeberle — COMMENTED** [review]
Agreeing with @martin-lysk on most points – LGTM 👍

File: lix/packages/client/src/git/status-list.test.ts [hunk 2]
```diff
@@ -219,7 +219,7 @@ describe(
 			// })
 
 			// it("*deleted and deleted status", async () => {
-			await repository.nodeishFs.readFile("./svelte.config.js", fileContent)
+			await repository.nodeishFs.readFile("./svelte.config.js")
```
[the remark this answers is not in the document]
Comment (janfjohannes) [reply lix/packages/client/src/git/status-list.test.ts:222]: good catch, this is from a time when tests were disabled from ts, but we should typecheck tests now too, as there were allready 2 important errors in tests that would have been catched by types

**Comment by felixhaeberle** [comment]
@janfjohannes whats blocking here?

**Comment by lucidNTR** [comment]
cc @felixhaeberle  

merge was delayed because of "conference" and me traveling. i someone should be available and have a look at possible bugs for three days after merge as path bugs are always annoying and can be time consuming and complete blockers for our users.

the dependent PRs are using this pr to test the path normalization in my other work before merging this, any bug we spot before this is on main is help for us but there has been none for 2 weeks.

**Comment by felixhaeberle** [comment]
Alright, so let's merge this on Monday:)

**Pull request merged**